config/config.go

```diff
@@ -1,281 +1,171 @@
+	Type string `mapstructure:"type"`
 
 	// directory to find specifications
-	Path string `yaml:"path" koanf:"path"`
+	Path string `mapstructure:"path"`
 
 	// backup configuration
-	Backup map[string]string `yaml:"backup" koanf:"backup"`
+	Backup map[string]string `mapstructure:"backup"`
 }
 
 type Job struct {
 	// directory to find specifications
-	Path string `yaml:"path"`
+	Path string `mapstructure:"path"`
 }
 
-type ProjectConfig struct {
-	// per project
-	Global map[string]string `yaml:"global"`
+type Project struct {
+	Name   string            `mapstructure:"name"`
+	Config map[string]string `mapstructure:"config"`
+}
 
-	// per namespace
-	Local map[string]string `yaml:"local"`
+type Namespace struct {
+	Name      string            `mapstructure:"name"`
+	Config    map[string]string `mapstructure:"config"`
+	Job       Job               `mapstructure:"job"`
+	Datastore []Datastore       `mapstructure:"datastore"`
 }
 
 type LogConfig struct {
 	// log level - debug, info, warning, error, fatal
-	Level string `yaml:"level"`
+	Level string `mapstructure:"level"`
 
 	// format strategy - plain, json
-	Format string `yaml:"format"`
+	Format string `mapstructure:"format"`
 }
 
 type ServerConfig struct {
 	// port to listen on
-	Port int `yaml:"port"`
+	Port int `mapstructure:"port" default:"9100"`
 	// the network interface to listen on
-	Host string `yaml:"host"`
+	Host string `mapstructure:"host" default:"0.0.0.0"`
 
 	// service ingress host for jobs to communicate back to optimus
-	IngressHost string `yaml:"ingress_host"`
+	IngressHost string `mapstructure:"ingress_host"`
 
 	// random 32 character hash used for encrypting secrets
-	AppKey string `yaml:"app_key"`
+	AppKey string `mapstructure:"app_key"`
 
-	DB                      DBConfig       `yaml:"db"`
-	Metadata                MetadataConfig `yaml:"metadata"`
-	ReplayNumWorkers        int            `yaml:"replay_num_workers"`
-	ReplayWorkerTimeoutSecs time.Duration  `yaml:"replay_worker_timeout_secs"`
-	ReplayRunTimeoutSecs    time.Duration  `yaml:"replay_run_timeout_secs"`
+	DB                  DBConfig       `mapstructure:"db"`
+	Metadata            MetadataConfig `mapstructure:"metadata"`
+	ReplayNumWorkers    int            `mapstructure:"replay_num_workers" default:"1"`
+	ReplayWorkerTimeout time.Duration  `mapstructure:"replay_worker_timeout" default:"120s"`
+	ReplayRunTimeout    time.Duration  `mapstructure:"replay_run_timeout"`
 }
 
 type DBConfig struct {
 	// database connection string
 	// e.g.: postgres://user:password@host:123/database?sslmode=disable
-	DSN string `yaml:"host"`
+	DSN string `mapstructure:"dsn"`
 
 	// maximum allowed idle DB connections
-	MaxIdleConnection int `yaml:"max_idle_connection"`
+	MaxIdleConnection int `mapstructure:"max_idle_connection" default:"5"`
 
 	// maximum allowed open DB connections
-	MaxOpenConnection int `yaml:"max_open_connection"`
+	MaxOpenConnection int `mapstructure:"max_open_connection" default:"10"`
```

Comment:
I think these numbers were changed to 20(open) and 5(idle)?

Reply:
hi @zakiindra, do you have context on this?

Reply:
also need to cross verify other default values